features/data/matching_banners_fingerprints.xml

```diff
@@ -8,10 +8,12 @@
     <param pos="1" name="pureftpd.config"/>
     <param pos="0" name="service.family" value="Pure-FTPd"/>
     <param pos="0" name="service.product" value="Pure-FTPd"/>
+    <param pos="0" name="service.protocol" value="ftp"/>
   </fingerprint>
   <fingerprint pattern="^(\S+) FTP Server \(SunOS (\S+)\) ready\.?$" flags="REG_ICASE">
     <description>SunOS/Solaris</description>
     <example>example.com FTP server (SunOS 5.7) ready.</example>
+    <param pos="0" name="service.protocol" value="ftp"/>
```

Comment:
The two individual fingerprints here still use `service.protocol`, overriding the same value now present in the `protocol` element in the top level of the document.  To put this another way, I was suggesting mixing the various ways you can specify the protocol to ensure that all ways result in the output fingerprints having the correct protocol.